common/optimiser/Optimiser.java

```diff
@@ -83,67 +110,76 @@ public boolean isError() {
         return status == Status.ERROR;
     }
 
-    public synchronized void restartFromSolution() {
-        if (hasSolver) {
-            solver.reset();
-            setSolverModel();
-        } else initialiseSolver();
+    public void setConstraintsChanged() {
+        isConstraintsUpToDate = true;
     }
 
-    private void initialiseSolver() {
+    private void createSolver() {
```

Comment:
I prefer `create` to `initialise` here because create/release are more like opposites. Also, `initialise` kind of has a connotation that it only happens once - initialisation normally occurs once per class. `create` is more flexible in the english language.